### PR TITLE
turbopack: Implement Server Actions from Client Components

### PR DESCRIPTION
**Pull request by jridgewell**
### What?

This completes Turbopack's Server Actions implementation!

It also cleans up several things:
- Removes `server_actions` configuration, it's enabled by default now.
- Fixes the transform that runs in the SSR layer (it actually needs to use the client transform)
- Generates a new encryption key

### How?

It crawls the client layer modules for additional Server Actions imports, and adds them to our generated `actions.js` entry point.

Closes NEXT-
Fixes #

Closes WEB-1854

### CODE REVIEW

File: packages/next-swc/crates/next-api/src/server_actions.rs [hunk 1]
```diff
@@ -171,10 +167,22 @@ async fn build_manifest(
 /// comment which identifies server actions. Every found server action will be
 /// returned along with the module which exports that action.
 #[turbo_tasks::function]
-async fn get_actions(module: Vc<Box<dyn Module>>) -> Result<Vc<ModuleActionMap>> {
+async fn get_actions(
+    rsc_entry: Vc<Box<dyn Module>>,
+    server_reference_modules: Vc<Vec<Vc<Box<dyn Module>>>>,
```
Comment (sokra) [inline packages/next-swc/crates/next-api/src/server_actions.rs:172]: Should we rename that?

File: packages/next-swc/crates/next-api/src/server_actions.rs [hunk 2]
```diff
@@ -192,11 +219,11 @@ async fn get_actions(module: Vc<Box<dyn Module>>) -> Result<Vc<ModuleActionMap>>
 /// Our graph traversal visitor, which finds the primary modules directly
 /// referenced by [parent].
 async fn get_referenced_modules(
-    parent: Vc<Box<dyn Module>>,
-) -> Result<impl Iterator<Item = Vc<Box<dyn Module>>> + Send> {
-    primary_referenced_modules(parent)
+    (layer, module): (ActionLayer, Vc<Box<dyn Module>>),
```
Comment (sokra) [inline packages/next-swc/crates/next-api/src/server_actions.rs:222]: Why isn't that just 2 arguments

**Review by sokra — DISMISSED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js jrl-client-server-actions | Change |
| - | - | - | - |
| buildDuration | 10.1s | 10.1s | N/A |
| buildDurationCached | 6s | 6s | N/A |
| nodeModulesSize | 175 MB | 175 MB | ⚠️ +486 B |
| nextStartRea..uration (ms) | 398ms | 394ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js jrl-client-server-actions | Change |
| - | - | - | - |
| 199-HASH.js gzip | 32.3 kB | 32.4 kB | N/A |
| 3f784ff6-HASH.js gzip | 53.2 kB | 53.2 kB | N/A |
| 99.HASH.js gzip | 182 B | 182 B | ✓ |
| framework-HASH.js gzip | 45.5 kB | 45.5 kB | ✓ |
| main-app-HASH.js gzip | 254 B | 252 B | N/A |
| main-HASH.js gzip | 35.3 kB | 35.4 kB | N/A |
| webpack-HASH.js gzip | 1.75 kB | 1.75 kB | N/A |
| Overall change | 45.7 kB | 45.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js jrl-client-server-actions | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js jrl-client-server-actions | Change |
| - | - | - | - |
| _app-HASH.js gzip | 206 B | 205 B | N/A |
| _error-HASH.js gzip | 182 B | 180 B | N/A |
| amp-HASH.js gzip | 506 B | 505 B | N/A |
| css-HASH.js gzip | 322 B | 323 B | N/A |
| dynamic-HASH.js gzip | 2.59 kB | 2.59 kB | ✓ |
| edge-ssr-HASH.js gzip | 260 B | 259 B | N/A |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 369 B | 369 B | ✓ |
| image-HASH.js gzip | 4.38 kB | 4.38 kB | N/A |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.67 kB | 2.67 kB | N/A |
| routerDirect..HASH.js gzip | 316 B | 318 B | N/A |
| script-HASH.js gzip | 385 B | 384 B | N/A |
| withRouter-HASH.js gzip | 319 B | 319 B | ✓ |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.99 kB | 3.99 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js jrl-client-server-actions | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 482 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js jrl-client-server-actions | Change |
| - | - | - | - |
| index.html gzip | 528 B | 530 B | N/A |
| link.html gzip | 542 B | 543 B | N/A |
| withRouter.html gzip | 524 B | 524 B | ✓ |
| Overall change | 524 B | 524 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js jrl-client-server-actions | Change |
| - | - | - | - |
| edge-ssr.js gzip | 96 kB | 96 kB | N/A |
| page.js gzip | 140 kB | 140 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js jrl-client-server-actions | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 622 B | 625 B | N/A |
| middleware-r..fest.js gzip | 150 B | 151 B | N/A |
| middleware.js gzip | 25.8 kB | 25.8 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 1.92 kB | 1.92 kB | ✓ |

</details>

</details>
Commit: 84251f1b81fa4a03e1b319ad1c4448bc77c46dbe

**Review by sokra — APPROVED** [review]

File: packages/next-swc/crates/next-api/src/server_actions.rs (hunk 2 above)
Comment (jridgewell) [reply packages/next-swc/crates/next-api/src/server_actions.rs:222]: This is invoked by `NonDeterministic::visit`'s `fn` impl, which only accepts a single arg.

**Pull request merged**